src/doc/grammar.md

```diff
@@ -780,6 +780,14 @@ never_type : "!" ;
 
 **FIXME:** grammar?
 
+### Type parameter bounds
+
+```antlr
+bound := ty_bound | lt_bound
+lt_bound := lifetime
+ty_bound := [?] [ for<lt_param_defs> ] simple_path
+```
+
```

Comment:
Could you also remove the bound grammar from `### Closure types` and move the `bound-list` grammar into this paragraph (adjusting it with optional trailing `+`).